person.rb

```diff
@@ -0,0 +1,30 @@
+require './nameable'
+
+# Parent class for teacher and student
+class Person < Nameable
+  attr_reader :id
+  attr_accessor :age, :name, :rentals
+
+  def initialize(age, name = 'Unknown', parent_permision: true)
+    super()
+    @id = Random.rand(1..100)
+    @name = name
+    @age = age
+    @rentals = []
+    _parent_permision = parent_permision
+  end
+
+  def can_use_services?
+    of_age? || _parent_permision
+  end
+
+  def correct_name
+    @name
+  end
+
+  private
+
+  def of_age?
+    age >= 18
+  end
+end
```

Comment:
- [ ] <b>REQUIRED:</b> As mentioned by the previous reviewer, you have add `add_rental` function in both `Book` and here in `Person` class. Think about it this way, when you create a rental, you will have to specify which book is rented and by who.  which means there is a `many-to-many` (also you can call it `has-many-through`) relationship between `Person` and `Book`. Because a person can rend more than one book and also Book can be rented by more than one people (after it is returned).  Therefore `Rental` is only acting as an intermediate class.

```rb
# add this function in Book
def add_rental(person, date)
# *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association`
   Rental.new(date, self, person)
end
# Add this one in Person
def add_rental(book, date)
# *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' #association.
  Rental.new(date, book, self)
end
```
Considering you have added those functions respectively. 
when you are going to create a rental, you will be asked to provide which book is rented and by who.
lets say you have 
   - This Book 
      ```rb
        book1 = Book.new('some title', 'some author', 'rentals' # current rentals #)
      ```
  -  And this Person 
      ```rb
        person1 = Person.new('some name', 'some age', 'rentals' # current rentals #) 
      ``` 
  -  So how do you get that person to rent that book
      -  you use `add_rental` function that is now in both of our classes `Book` and `Person`.
         ```rb
           book1.add_rental(person1, Date.new) # This will initialize new rental with person1 and book its self
           person1.add_rental(book1, Date.new) # This will initialize new rental with book1 and Person him/her self
         ```